src/goto-instrument/contracts/instrument_spec_assigns.cpp

```diff
@@ -27,8 +27,62 @@ Date: January 2022
 
 #include "utils.h"
 
-///// PUBLIC METHODS /////
+void add_pragma_disable_pointer_checks(source_locationt &location)
+{
+  location.add_pragma("disable:pointer-check");
+  location.add_pragma("disable:pointer-primitive-check");
+  location.add_pragma("disable:pointer-overflow-check");
+  location.add_pragma("disable:signed-overflow-check");
+  location.add_pragma("disable:unsigned-overflow-check");
+  location.add_pragma("disable:conversion-check");
+}
+
+goto_programt::instructiont &
+add_pragma_disable_pointer_checks(goto_programt::instructiont &instr)
+{
+  add_pragma_disable_pointer_checks(instr.source_location_nonconst());
+  return instr;
+}
 
+goto_programt &add_pragma_disable_pointer_checks(goto_programt &prog)
+{
+  Forall_goto_program_instructions(it, prog)
+    add_pragma_disable_pointer_checks(*it);
+  return prog;
+}
+
+/// a local pragma used to keep track and skip already instrumented instructions
+#define CONTRACT_PRAGMA_DISABLE_ASSIGNS_CHECK "contracts:disable:assigns-check"
```

Comment:
```
/// A local pragma used to keep track and skip already instrumented instructions.
```

I'd also place all definitions right after includes.

Reply:
made it a `const std::string` placed at the top